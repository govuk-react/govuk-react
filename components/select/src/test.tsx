import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';

import { Select } from '.';

describe('Select', () => {
  const example = 'example';
  const meta = {
    touched: true,
    error: example,
  };
  const wrapper = <Select label={example}>{example}</Select>;

  const wrapperErrorText = (
    <Select hint={example} errorText={example} meta={meta} label={example}>
      {example}
    </Select>
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('matches snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('enzyme.mount');
  });

  it('matches snapshot for errorText', () => {
    expect(mount(wrapperErrorText)).toMatchSnapshot('errorText mount');
  });
});
