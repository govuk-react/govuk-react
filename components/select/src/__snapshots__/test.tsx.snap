// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select matches snapshot for errorText: errorText mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  border-left: 4px solid #d4351c;
  margin-right: 15px;
  padding-left: 10px;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-bottom: 15px;
  clear: both;
  color: #d4351c;
}

.c2 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-bottom: 15px;
  color: #6f777b;
}

.c4 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 33px;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
  border: 4px solid #d4351c;
}

.c4:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media print {
  .c3 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c2 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c4 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    width: 50%;
    height: 38px;
  }
}

<Select
  errorText="example"
  hint="example"
  input={Object {}}
  label="example"
  meta={
    Object {
      "error": "example",
      "touched": true,
    }
  }
>
  <Label
    error={true}
    errorText="example"
  >
    <src__StyledLabel
      error={true}
      errorText="example"
    >
      <label
        className="c0"
      >
        <LabelText>
          <src__StyledLabelText>
            <span
              className="c1"
            >
              example
            </span>
          </src__StyledLabelText>
        </LabelText>
        <HintText>
          <src__StyledHint>
            <span
              className="c2"
            >
              example
            </span>
          </src__StyledHint>
        </HintText>
        <ErrorText>
          <src__StyledError>
            <span
              className="c3"
            >
              example
            </span>
          </src__StyledError>
        </ErrorText>
        <src__StyledSelect
          error="example"
        >
          <select
            className="c4"
          >
            example
          </select>
        </src__StyledSelect>
      </label>
    </src__StyledLabel>
  </Label>
</Select>
`;

exports[`Select matches snapshot: enzyme.mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c2 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 33px;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
}

.c2:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media print {
  .c2 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    width: 50%;
    height: 38px;
  }
}

<Select
  input={Object {}}
  label="example"
  meta={Object {}}
>
  <Label
    error={false}
  >
    <src__StyledLabel
      error={false}
    >
      <label
        className="c0"
      >
        <LabelText>
          <src__StyledLabelText>
            <span
              className="c1"
            >
              example
            </span>
          </src__StyledLabelText>
        </LabelText>
        <src__StyledSelect>
          <select
            className="c2"
          >
            example
          </select>
        </src__StyledSelect>
      </label>
    </src__StyledLabel>
  </Label>
</Select>
`;
